@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -404,6 +428,46 @@ export class TextElementsRenderer {
         this.m_options.delayLabelsUntilMovementFinished = delay;
     }
 
+    /**
+     * @internal
+     * @hidden
+     *
+     * This is exposed for the usage in testing only
+     */
+    set textCanvasFactory(textCanvasFactory: TextCanvasFactory) {
+        this.m_textCanvasFactory = textCanvasFactory;
+    }
+
+    /**
+     * @internal
+     * @hidden
+     *
+     * This is exposed for the usage in testing only
+     */
+    set poiRenderer(poiRenderer: PoiRenderer) {
+        this.m_poiRenderer = poiRenderer;
+    }
+
+    /**
+     * @internal
+     * @hidden
+     *
+     * This is exposed for the usage in testing only
+     */
+    get screenCollisions(): ScreenCollisions {
+        return this.m_screenCollisions;
+    }
+
+    /**
+     * @internal
+     * @hidden
+     *
+     * This is exposed for the usage in testing only
+     */
+    set screenCollisions(screenCollisions: ScreenCollisions) {
+        this.m_screenCollisions = screenCollisions;
+    }
```

Comment:
An alternative would be to have a static method to build the TextElementsRenderer for production, where we create these objects and pass them to the constructor.
The tests would call directly the constructor to inject their own implementations for text canvas factor, poi renderer ...

Reply:
I moved them as optional parameters to the constructor